Add Dangerfile

aamctustwo opened the pull request:
Resolves #11 `Add Dangerfile`.
### Summary

Adds a Dangerfile for PR linting.
### Details
- Adds Dangerfile and adds `danger` command to Fastfile.
- Adjust `verify_demo` lane to only build the demo app but not run UI Tests, as they have been flakey on Travis CI.
- Update SwiftLint configuration (as it was using old rule names).
### Other Notes
- Currently will not work with Forks due to using encrypted environment variables for Travis CI.

Commit:
Add Dangerfile

Commit:
Build demo app instead of running UI Tests

as the UI tests have been extremely flakey on Travis CI

Commit:
Ensure SwiftLint is installed on Travis CI

Commit:
Add `danger-swiftlint` gem

ghost commented:
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="true">PR is classed as Work in Progress</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

Commit:
Force only linting of the Sources folder

Commit:
Add attribution to Dangerfile

Commit:
Clean up SwiftLint configuration

- Enable `type_body_length` and `variable_name`
- Fully disable `line_length`
- Use defaults for `variable_name`
